lib/manifest.js

```diff
@@ -0,0 +1,78 @@
+let path = require("path");
+let { createFile, generateFingerprint, uriJoin } = require("faucet-pipeline-util");
+let retry = require("./retry");
+let retryTimings = [10, 50, 100, 250, 500, 1000];
+
+class Manifest {
+	constructor(config, fingerprint, configDir, exitOnError) {
```

Comment:
* passing through an unspecified `config` object is most unhelpful; we should be explicit in what we expect here
* we should relegate `fingerprint` and `exitOnError` to an `options` object:

  ```
  constructor(config, configDir, { fingerprint, exitOnError })
  ```

  That both clarifies intent and is more future-proof.